appsec/src/helper/subscriber/waf.cpp

```diff
@@ -644,8 +644,16 @@ void instance::listener::call(
                     auto value =
                         static_cast<double>(static_cast<uint64_t>(derivative));
                     metrics_attributes_.emplace(derivative.key(), value);
+                } else if (derivative.is_float()) {
+                    auto value = static_cast<double>(derivative);
+                    metrics_attributes_.emplace(derivative.key(), value);
                 } else {
-                    meta_attributes_.emplace(derivative.key(), derivative);
+                    if (derivative.is_string()) {
+                        meta_attributes_.emplace(derivative.key(), derivative);
+                    } else {
+                        SPDLOG_WARN("Schema for key {} is not a valid type",
+                            derivative.key());
+                    }
```

Comment:
I meant something along the lines of 

```cpp
} else if (derivative.is_string()) {
   ...
} else {
  // log error
}
```

Also why does the warning say `Schema for key`? I'd change to "Unsupported attribute type for key {}"